espnet/asr/pytorch_backend/asr.py

```diff
@@ -514,16 +514,23 @@ def train(args):
     elif args.opt == "noam":
         from espnet.nets.pytorch_backend.transformer.optimizer import get_std_opt
 
-        # For transformer-transducer, adim declaration is within the block definition.
-        # Thus, we need retrieve the most dominant value (d_hidden) for Noam scheduler.
-        if hasattr(args, "enc_block_arch") or hasattr(args, "dec_block_arch"):
-            adim = model.most_dom_dim
+        if "transducer" in mtl_mode:
```

Comment:
Can you make these to be minimum changes?
I see that the name of `args.transformer_***` does not make sense, but the inconsistent change (only apply this change in the transducer mode) becomes more inconsistent.

Reply:
My apologies, I'm a bit confused. Do you want me to revert that part?

> Remove scheme to automatically find most dominant dimension for Noam scheduler and rely on newly added noam-adim argument instead. Noam can now be used with RNN architecture.

For `args.transformer_***`, I added the old parameters with a deprecation notice as requested: https://github.com/espnet/espnet/pull/3217/commits/38dc2bc9236abc9973bd7fd085d5f7bb4189aea4
In relation to the part you pointed out, both `adim` and new `noam-adim` are supported.

Reply:
Yes, I noticed it, but I feel that keeping the different naming does not look good to me and the deprecation notice is not enough.